src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -311,6 +311,11 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
     private fun checkImpl(holder: AnnotationHolder, impl: RsImplItem) {
         val trait = impl.traitRef?.resolveToTrait ?: return
         val traitName = trait.name ?: return
+        if (impl.unsafe != null && trait.unsafe == null) {
+            holder.createErrorAnnotation(impl.traitRef!!, "Safe traits should not have unsafe implementations [E0199]")
+        } else if (impl.unsafe == null && trait.unsafe != null) {
+            holder.createErrorAnnotation(impl.traitRef!!, "Unsafe traits must have unsafe implementations [E0200]")
```

Comment:
Let's declare `traitRef = impl.traitRef ?: return` at the very method beginning and get rid of these `!!`s.